### ISSUE TITLE
Temperature scaling

### ISSUE THREAD

**Issue by Imartinezcuevas** [Imartinezcuevas/nano-llm#12]
Apply temperature scaling to logits before softmax to control randomness.
